airbyte-integrations/connectors/source-lever/source_lever/source.py

```diff
@@ -0,0 +1,267 @@
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "archive_reasons"
+
+
+# TODO: Basic incremental stream
+class IncrementalLeverStream(LeverStream, ABC):
+    """
+    TODO fill in details of this class to implement functionality related to incremental syncs for your connector.
+         if you do not need to implement incremental sync for any streams, remove this class.
+    """
+
+    # TODO: Fill in to checkpoint stream reads after N records. This prevents re-reading of data if the stream fails for any reason.
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        """
+        TODO
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
+
+
+# Basic Sub streams using Opportunity id
+class OpportunitySubStream(LeverStream, ABC):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        
+    # def __init__(self, start_date: str, **kwargs):
+    #     super().__init__(**kwargs)
+    #     self._start_date = start_date
+    
+    def path(self, stream_slice: Mapping[str, any] = None, **kwargs) -> str:
+        return f"opportunities/{stream_slice['opportunity_id']}/{self.name}"
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        for stream_slice in super().stream_slices(**kwargs):
+            # opportunities_stream = Opportunities(authenticator=self.authenticator, base_url=self.base_url, start_date=self._start_date)
+            opportunities_stream = Opportunities(authenticator=self.authenticator)
+            for opportunity in opportunities_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice):
+                yield {"opportunity_id": opportunity["id"]}
+    
+
+    def parse_response(self, response: requests.Response, stream_slice:[Mapping[str, Any]], **kwargs) -> Iterable[Mapping]:
+        records = response.json()["data"]
+        if not records:
+            records = [{}]
+        
+        for record in records:
+            record["opportunity"] = stream_slice["opportunity_id"]
+        yield from records
```

Comment:
The fix for causing heartbeat issues.

Reply:
Can you add a comment explaining the fix?